be/src/exec/pipeline/analysis/analytic_sink_operator.h

```diff
@@ -29,10 +29,13 @@ class AnalyticSinkOperator : public Operator {
     Status push_chunk(RuntimeState* state, const vectorized::ChunkPtr& chunk) override;
 
 private:
-    Status _process_by_partition_if_necessary();
+    Status _process_by_partition_if_necessary_for_other();
+    Status _process_by_partition_if_necessary_for_rows_between_unbounded_preceding_and_following();
+    Status (AnalyticSinkOperator::*_process_by_partition_if_necessary)() = nullptr;
```

Comment:
how about using std::function instead of function pointer? it's more clearly

Reply:
std::function performance is bad

Reply:
Good idea, i tyr

Reply:
@kangkaisen _process_by_partition_if_necessary is not called very often, so there are no significant performance issues.

Reply:
https://stackoverflow.com/questions/5057382/what-is-the-performance-overhead-of-stdfunction

Reply:
it's not easy to evaluate the impact on performance, so i currently don't modify it now